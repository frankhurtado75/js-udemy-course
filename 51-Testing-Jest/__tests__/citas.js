import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', () => {
    
    const citas = new Citas();

    const id = Date.now();
    
    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Bob',
            propietario: 'Frank',
            telefono: '0414',
            fecha: '10-12-2000',
            hora:'10:30',
            sintomas: 'Tos'
        };

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });
    
    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo mascota',
            propietario: 'Frank',
            telefono: '0414',
            fecha: '10-12-2000',
            hora:'10:30',
            sintomas: 'Tos'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
});