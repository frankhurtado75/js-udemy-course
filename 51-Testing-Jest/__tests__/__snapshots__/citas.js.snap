// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2000",
      "hora": "10:30",
      "id": 1620610949403,
      "mascota": "Nuevo mascota",
      "propietario": "Frank",
      "sintomas": "Tos",
      "telefono": "0414",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2000",
      "hora": "10:30",
      "id": 1620610949403,
      "mascota": "Bob",
      "propietario": "Frank",
      "sintomas": "Tos",
      "telefono": "0414",
    },
  ],
}
`;
